seatunnel-connectors-v2/connector-hive/src/main/java/org/apache/seatunnel/connectors/seatunnel/hive/sink/HiveSinkAggregatedCommitter.java

```
@@ -28,41 +34,82 @@
+                try {
+                    for (Map.Entry<String, Map<String, String>> entry : aggregateCommitInfo.getFileAggregatedCommitInfo().getTransactionMap().entrySet()) {
+                        // rollback the file
+                        for (Map.Entry<String, String> mvFileEntry : entry.getValue().entrySet()) {
+                            HdfsUtils.renameFile(mvFileEntry.getKey(), mvFileEntry.getValue(), true);
+                        }
+                        // delete the transaction dir
+                        HdfsUtils.deleteFile(entry.getKey());
+                    }
+                    // add hive partition
+                    aggregateCommitInfo.getFileAggregatedCommitInfo().getPartitionDirAndValsMap().entrySet().forEach(entry -> {
+                        Partition part = new Partition();
+                        part.setDbName(aggregateCommitInfo.getTable().getDbName());
+                        part.setTableName(aggregateCommitInfo.getTable().getTableName());
+                        part.setValues(entry.getValue());
+                        part.setParameters(new HashMap<>());
+                        part.setSd(aggregateCommitInfo.getTable().getSd().deepCopy());
+                        part.getSd().setSerdeInfo(aggregateCommitInfo.getTable().getSd().getSerdeInfo());
+                        part.getSd().setLocation(aggregateCommitInfo.getTable().getSd().getLocation() + "/" + entry.getKey());
+                        try {
+                            hiveMetaStoreClient.add_partition(part);
+                        } catch (TException e) {
+                            throw new RuntimeException(e);
+                        }
+                    });
+                } catch (Exception e) {
+                    LOGGER.error("commit aggregateCommitInfo error ", e);
+                    errorAggregatedCommitInfoList.add(aggregateCommitInfo);
                 }
```

It would be better to throw an exception directly here, and the upper layer should capture the exception and encapsulate the `errorAggregatedCommitInfoList` (if is necessary).